### PR TITLE
[NEW] New sidebar layout

### LINKED ISSUE #9541: New sidebar header and views
- [x] Header
- [x] Search
- [x] Sidebar views
	- [x] Extended view
	- [x] Medium view
	- [x] Condensed view
- [x] Sort and filters
	- [x] Alphabetical sort
	- [x] Activity sort
	- [x] Favorites filter
	- [x] Unreads filter
- [x] Tests and review

RocketChat will offer a new sidebar experience with a new header, 'extended', 'medium' and 'condensed' view and a sort/filter system.

image

### PR DESCRIPTION
**Pull request by ggazzo**
image
image
image

@RocketChat/core 

Close #9541

closes https://github.com/RocketChat/Rocket.Chat/issues/9224

### CODE REVIEW

**Comment by sampaiodiego** [comment]
screenshots pls

**Comment by ggazzo** [comment]
@sampaiodiego 
flw3jipl3c

File: packages/rocketchat-lib/lib/roomTypes/private.js [hunk 1]
```diff
@@ -44,9 +44,9 @@ export class PrivateRoomType extends RoomTypeConfig {
 
 	condition() {
 		const user = Meteor.user();
-		const roomsListExhibitionMode = RocketChat.getUserPreference(user, 'roomsListExhibitionMode');
+		// const roomsListExhibitionMode = RocketChat.getUserPreference(user, 'roomsListExhibitionMode');
```
Comment (gdelavald) [inline packages/rocketchat-lib/lib/roomTypes/private.js]: Remove comments.

File: packages/rocketchat-theme/client/imports/components/sidebar/sidebar-item.css [hunk 2]
```diff
@@ -243,12 +247,18 @@
 
 		& * {
 			display: inline !important;
+
```
Comment (gdelavald) [inline packages/rocketchat-theme/client/imports/components/sidebar/sidebar-item.css]: Is there any way we can remove this !important statements?

File: packages/rocketchat-ui-account/client/accountPreferences.html [hunk 3]
```diff
@@ -101,76 +101,76 @@ <h1>{{_ "Messages"}}</h1>
...
 								<label>{{_ "Hide_Avatars"}}</label>
 								<div>
-									<label><input type="radio" name="hideAvatars" value="1" checked="{{checked 'hideAvatars' true}}"/> {{_ "True"}}</label>
-									<label><input type="radio" name="hideAvatars" value="0" checked="{{checked 'hideAvatars' false}}"/> {{_ "False"}}</label>
+									<label><input type="radio" name="hideAvatars" value="true" checked="{{checked 'hideAvatars' true}}"/> {{_ "True"}}</label>
+									<label><input type="radio" name="hideAvatars" value="false" checked="{{checked 'hideAvatars' false}}"/> {{_ "False"}}</label>
 								</div>
 							</div>
-							<div class="input-line double-col">
+							<!-- <div class="input-line double-col">
```
Comment (gdelavald) [inline packages/rocketchat-ui-account/client/accountPreferences.html]: Remove html comments

File: packages/rocketchat-ui-sidenav/client/roomList.js [hunk 4]
```diff
@@ -5,48 +5,58 @@ import { UiTextContext } from 'meteor/rocketchat:lib';
...
 	rooms() {
-		if (this.identifier === 'unread') {
-			const query = {
-				alert: true,
-				open: true,
-				hideUnreadStatus: {$ne: true}
-			};
-			return ChatSubscription.find(query, {sort: {'t': 1, 'name': 1}});
-		}
+
```
Comment (gdelavald) [inline packages/rocketchat-ui-sidenav/client/roomList.js]: Remove comments.

File: packages/rocketchat-ui-sidenav/client/sidebarHeader.js [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+											popover.close();
+										}
+									},
+									{
+										icon: 'sign-out',
+										name: t('Logout'),
+										type: 'open',
+										id: 'logout',
+										action: () => {
+											// const user = Meteor.user();
```
Comment (gdelavald) [inline packages/rocketchat-ui-sidenav/client/sidebarHeader.js]: Comment

File: tests/end-to-end/ui/09-channel.js [hunk 6]
```diff
@@ -22,14 +22,15 @@ describe('[Channel]', ()=> {
...
 	});
 	describe('[Search]', ()=> {
-		describe('[SpotlightSearch]', () => {
+		describe.skip('[SpotlightSearch]', () => {
 			describe('rocket.cat:', () => {
 
-				afterEach(() => {
-					sideNav.spotlightSearch.setValue('');
-				});
+				// after(() => {
```
Comment (gdelavald) [inline tests/end-to-end/ui/09-channel.js]: comment

File: tests/end-to-end/ui/09-channel.js [hunk 7]
```diff
@@ -41,15 +42,17 @@ describe('[Channel]', ()=> {
 			});
 
 			describe('general:', () => {
-				beforeEach(() => {
-					sideNav.getChannelFromSpotlight('general').waitForVisible(5000);
-				});
+				// beforeEach(() => {
```
Comment (gdelavald) [inline tests/end-to-end/ui/09-channel.js]: remove or skip.

**Comment by janrudolph** [comment]
I really like this new sidebar! 

@ggazzo @karlprieb @gdelavald 
Why not open the user related stuff (setting status, editing profile) by clicking on the profile picture? This would group the user related items. The burger menu could then be used for all other options, like opening the admin interface (or any future options). If there are no other options, the burger menu will disappear. This might look cleaner.

File: packages/rocketchat-theme/client/imports/components/sidebar/sidebar-item.css (hunk 2 above)
Comment (karlprieb) [reply packages/rocketchat-theme/client/imports/components/sidebar/sidebar-item.css]: No :(

File: packages/rocketchat-ui-account/client/accountPreferences.html (hunk 3 above)
Comment (karlprieb) [reply packages/rocketchat-ui-account/client/accountPreferences.html]: We will use this html when we implement that feature

**Comment by gdelavald** [comment]
@karlprieb noticed this error below, any chance we could take a look at that before merging?
```
TypeError: Cannot read property 'split' of null
    at isRtl (https://rocket-chat-pr-9608.herokuapp.com/a8f0980781712f74f939bb6c83765f38756814a7.js?meteor_js_resource=true:209:62117)
    at f.TemplateInstance.<anonymous> (https://rocket-chat-pr-9608.herokuapp.com/a8f0980781712f74f939bb6c83765f38756814a7.js?meteor_js_resource=true:199:9760)
```

**Comment by geekgonecrazy** [comment]
> Why not open the user related stuff (setting status, editing profile) by clicking on the profile picture? This would group the user related items. The burger menu could then be used for all other options, like opening the admin interface (or any future options). If there are no other options, the burger menu will disappear. This might look cleaner.

I agree.  When trying this out, to change the status my mouse went to click on my avatar to change my status(because that's what other apps including rocket.chat have trained me to do).  I think it would be more intuitive there.  I think we associate the status icon there with clicking it to change our status.

@karlprieb @arthurgiustina @engelgabriel thoughts?

**Comment by karlprieb** [comment]
Hey @janrudolph @geekgonecrazy we will merge this PR as planned, just like the screenshots.

I'll talk to @engelgabriel and @arthurgiustina about these changes and if they approve we will release on the next release.

File: packages/rocketchat-theme/client/imports/components/sidebar/sidebar-item.css [hunk 8]
```diff
@@ -142,7 +142,7 @@
 		width: var(--sidebar-item-user-status-size);
 		height: var(--sidebar-item-user-status-size);
 
-		margin: 0 7px 0 3px;
+		margin: 0 7px 0 7px;
```
Comment (ggazzo) [inline packages/rocketchat-theme/client/imports/components/sidebar/sidebar-item.css]: margin: 0 7px;

File: packages/rocketchat-theme/client/imports/general/base.css [hunk 9]
```diff
@@ -118,7 +118,7 @@ button {
 .rc-icon {
 	overflow: hidden;
 
-	width: 1em;
+	width: 1.25em;
```
Comment (ggazzo) [inline packages/rocketchat-theme/client/imports/general/base.css]: ?? width and height should have same value

**Comment by karlprieb** [comment]
@sampaiodiego 

image

image

image

and fixed search bar too :)

**Comment by johnyb0y** [comment]
@ggazzo I really like the changes. For admins with less tech savy users there's a problem though:
I rolled out 0.62.0-rc2 on our test server and the sidebar layout of existing users changed significantly.
It would be great if the standard layout of the new sidebar with its configurable options could be close to the look of older versions. 
Or even better: if you could roll out a standard setting for every user via the admin panel.
Thanks for your great work!

**Comment by reza899** [comment]
in RTL language (farsi in my case) the menu for user not apear correctly.
capture
in "my account" section, it may not work save changes. after chaning some option, it is not saved.

**Comment by karlprieb** [comment]
Thank you @reza899 for your report, can you open an issue for that?

**Comment by reza899** [comment]
yes, I opened it at https://github.com/RocketChat/Rocket.Chat/issues/10018

**Comment by eniris** [comment]
Hi @karlprieb I can't see the extended link in view mode list.
Is there is a setting ?
image

**Comment by karlprieb** [comment]
@eniris Great! 👍

**Pull request merged**